.github/.check_scopes.lua

```diff
@@ -0,0 +1,49 @@
+-- Copyright Cartesi Pte. Ltd.
+-- 
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+-- this file except in compliance with the License. You may obtain a copy of the
+-- License at http://www.apache.org/licenses/LICENSE-2.0
+-- 
+-- Unless required by applicable law or agreed to in writing, software distributed
+-- under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+-- CONDITIONS OF ANY KIND, either express or implied. See the License for the
+-- specific language governing permissions and limitations under the License.
+
+function extractScopesFromFile(filename)
+  local scopes = {}
+  local file = io.open(filename)
+  if not file then
+    error("Failed to open file: " .. filename)
+  end
+  for line in file:lines() do
+    local scope = line:match("%S+")
+    if scope and scopes[scope] == nil then
```

Comment:
Rather than just checking if `scopes[scope] == nil`, we should assert that this is not the case, because it would indicate that something is wrong with the `.scopes.txt` file.

Reply:
Fair enough. Either way, checking if `scopes[scope] == nil` is useless if the purpose is just to collect the set of valid scopes from the scopes file.